src/main/java/cuchaz/enigma/gui/EnigmaSyntaxKit.java

```diff
@@ -1,44 +1,52 @@
-        configuration.put("Style.IDENTIFIER", Config.getInstance().identifierColor + ", 0");
-        configuration.put("Style.DEFAULT", Config.getInstance().defaultTextColor + ", 0");
-        configuration.put(LineNumbersRuler.PROPERTY_BACKGROUND, Config.getInstance().lineNumbersBackground + "");
-        configuration.put(LineNumbersRuler.PROPERTY_FOREGROUND, Config.getInstance().lineNumbersForeground + "");
-        configuration.put(LineNumbersRuler.PROPERTY_CURRENT_BACK, Config.getInstance().lineNumbersSelected + "");
-        configuration.put("RightMarginColumn", "999"); //No need to have a right margin, if someone wants it add a config
-
-        configuration.put("Action.quick-find", "cuchaz.enigma.gui.QuickFindAction, menu F");
-    }
-
-    public static void invalidate(){
-        configuration = null;
-    }
+
+	private static Configuration configuration = null;
+
+	@Override
+	public Configuration getConfig() {
+		if (configuration == null) {
+			initConfig(DefaultSyntaxKit.getConfig(JavaSyntaxKit.class));
+		}
+		return configuration;
+	}
+
+	public void initConfig(Configuration baseConfig) {
+		configuration = baseConfig;
+		//See de.sciss.syntaxpane.TokenType
+		configuration.put("Style.KEYWORD", Config.getInstance().highlightColor + ", 0");
+		configuration.put("Style.KEYWORD2", Config.getInstance().highlightColor + ", 3");
+		configuration.put("Style.STRING", Config.getInstance().stringColor + ", 0");
+		configuration.put("Style.STRING2", Config.getInstance().stringColor + ", 1");
+		configuration.put("Style.NUMBER", Config.getInstance().numberColor + ", 1");
+		configuration.put("Style.OPERATOR", Config.getInstance().operatorColor + ", 0");
+		configuration.put("Style.DELIMITER", Config.getInstance().delimiterColor + ", 1");
+		configuration.put("Style.TYPE", Config.getInstance().typeColor + ", 2");
+		configuration.put("Style.TYPE2", Config.getInstance().typeColor + ", 1");
+		configuration.put("Style.IDENTIFIER", Config.getInstance().identifierColor + ", 0");
+		configuration.put("Style.DEFAULT", Config.getInstance().defaultTextColor + ", 0");
+		configuration.put(LineNumbersRuler.PROPERTY_BACKGROUND, Config.getInstance().lineNumbersBackground + "");
+		configuration.put(LineNumbersRuler.PROPERTY_FOREGROUND, Config.getInstance().lineNumbersForeground + "");
+		configuration.put(LineNumbersRuler.PROPERTY_CURRENT_BACK, Config.getInstance().lineNumbersSelected + "");
+		configuration.put("RightMarginColumn", "999"); //No need to have a right margin, if someone wants it add a config
+
+		configuration.put("Action.quick-find", "cuchaz.enigma.gui.QuickFindAction, menu F");
+
+		// This is an action written in javascript that is useless for enigma's
+		// use case, and removing it causes the editor to load way faster the
+		// first time
+		configuration.remove("Action.insert-date");
+	}
+
+	public static void invalidate() {
+		configuration = null;
+	}
+
```

Comment:
What happened here?

Reply:
Reformatted to use tabs like the rest of the code, I think.

Reply:
I think most of the new classes that have been added use spaces, though. We should probably decide on one format and reformat everything.

Reply:
Alright, then I'll revert the formatting to spaces for this class. I'll probably not modify this class anymore anyway (IDEA replaces spaces with tabs when formatting)

Reply:
I would suggest using checkstyle with almost identical rules to fabric and loom.